lib/models/movie_detail_model.dart

```diff
@@ -0,0 +1,15 @@
+import 'genre.dart';
+
+class MovieDetailModel {
+  final String title, posterPath, overview;
+  final List<Genre> genres;
+  final int id;
+  MovieDetailModel.fromJson(Map<String, dynamic> json)
+      : id = json['id'],
+        title = json['title'],
+        posterPath = json['poster_path'],
+        genres = (json['genres'] as List)
+            .map((genre) => Genre(id: genre['id'], name: genre['name']))
+            .toList(),
```

Comment:
```suggestion
        genres = json['genres']
            .map<String>((genre) => genre["name"].toString())
            .toList(),
```
`as List` 와 같은 타입 캐스팅의 경우 런타임에 에러가 날 가능성이 생기므로 최대한 지양하는 것이 좋습니다.

`clasa Genre` 로 감싸려는 시도는 좋았으나 다른 파일에서도 id 필드는 사용하고 있지 않습니다.
단순히 map 내에서 deserialize 를 처리하기 위함 이었다면 String 으로 쓰는 것이 나아보입니다.


단, 모든 원시값을 class 로 포장하겠다는 의도였다고 한다면 다른 존재하는 모든 String 들도 감싸주는 것이 적절합니다.
https://velog.io/@jhp1115/%EC%9B%90%EC%8B%9C%EA%B0%92%EA%B3%BC-%EB%AC%B8%EC%9E%90%EC%97%B4-%ED%8F%AC%EC%9E%A5